imports/plugins/core/router/lib/hooks.js

```diff
@@ -35,24 +35,24 @@ const Hooks = {
     }
   },
 
-  onEnter(routeName, callback, ...args) {
+  onEnter(routeNameOrGlobalCallback, callback) {
     // global onEnter callback
-    if (arguments.length === 1 && typeof args[0] === "function") {
```

Comment:
`args[0]` is the bug. in this case, since there are two named parameters, `args[0]` === `arguments[2]` which is always undefined. Since we name the parameters, there is no need to use the `args` array

Reply:
I think this came about as trying to fix `prefer rest over arguments` linting.